picard/ui/coverartbox.py

```diff
@@ -545,6 +550,43 @@ def load_remote_image(self, url, data):
             log.warning("Can't load image: %s", e)
             return
 
+    def delete_cover_art(self):
+        if not self.item or not self.item.metadata.images:
+            return
+
+        cover_art_list = [image.source or _("Unnamed Cover Art") for image in self.item.metadata.images]
+
+        selected_item, ok_pressed = QtWidgets.QInputDialog.getItem(self, _("Delete Cover Art"),
+                                                                    _("Select the cover art image to delete:"),
+                                                                    cover_art_list, 0, False)
+        if ok_pressed:
+            selected_image_index = cover_art_list.index(selected_item)
+            selected_image = self.item.metadata.images[selected_image_index]
+            if not isinstance(self.item, File):
+                self.delete_cover_art_for_item(self.item, selected_image)
+            for file in self.get_files_to_process():
+                self.delete_cover_art_for_item(file, selected_image)
+                self.update()
+            self.item.update()
+
+    def get_files_to_process(self):
+        if isinstance(self.item, (Album, FileListItem)):
+            return self.item.iterfiles()
+        elif isinstance(self.item, File):
+            return [self.item]
+
+    def delete_cover_art_for_item(self, item, selected_image):
```

Comment:
This method is definitively too simple.

Basically you should have something like https://github.com/metabrainz/picard/blob/8af19d40cd3f2e411808f41d6ea0768646c2f38f/picard/ui/coverartbox.py#L563-L600 but instead of `set_image_*()` calling `image_delete`.

The code has to prevent signals to be sent over and over (hence `enable_update_metadata_images(False)` and `update(signal=False)` in linked code.